src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1591,10 +1596,12 @@ void TorrentHandle::handleTorrentFinishedAlert(const libtorrent::torrent_finishe
         m_moveFinishedTriggers.append(boost::bind(&Session::handleTorrentFinished, m_session, this));
     }
     else {
-        if (recheckTorrentsOnCompletion)
+        if (recheckTorrentsOnCompletion && !m_didCheck)
```

Comment:
You also need to add this condition above to prevent deferred call to "forceRecheck()".

Reply:
done